caravel/views.py

```diff
@@ -901,130 +941,89 @@ def explore(self, datasource_type, datasource_id, slice_id=None):
-        if not viz_type and datasource.default_endpoint:
-            return redirect(datasource.default_endpoint)
-
-        # default to table if no default endpoint and no viz_type
-        if not viz_type:
-            viz_type = "table"
-
-        # validate viz params
-        try:
-            obj = viz.viz_types[viz_type](
-                datasource,
-                form_data=slice_params,
-                slice_=slc)
-        except Exception as e:
-            flash(utils.error_msg_from_exception(e), "danger")
-            return redirect(error_redirect)
+                slice_params_multi_dict, slc, slice_add_perm, slice_edit_perm)
 
         # handle different endpoints
-        if slice_params.get("json") == "true":
-            status = 200
+        if slice_params_multi_dict.get("json") == "true":
             if config.get("DEBUG"):
                 # Allows for nice debugger stack traces in debug mode
-                payload = obj.get_json()
-            else:
-                try:
-                    payload = obj.get_json()
-                except Exception as e:
-                    logging.exception(e)
-                    payload = utils.error_msg_from_exception(e)
-                    status = 500
-            resp = Response(
-                payload,
-                status=status,
-                mimetype="application/json")
-            return resp
+                return Response(
+                    viz_obj.get_json(),
+                    status=200,
+                    mimetype="application/json")
+            try:
+                return Response(
+                    viz_obj.get_json(),
+                    status=200,
+                    mimetype="application/json")
+            except Exception as e:
+                logging.exception(e)
+                return json_error_response(utils.error_msg_from_exception(e))
 
-        elif slice_params.get("csv") == "true":
-            payload = obj.get_csv()
+        elif slice_params_multi_dict.get("csv") == "true":
+            payload = viz_obj.get_csv()
             return Response(
                 payload,
                 status=200,
                 headers=generate_download_headers("csv"),
                 mimetype="application/csv")
         else:
-            if slice_params.get("standalone") == "true":
+            if slice_params_multi_dict.get("standalone") == "true":
                 template = "caravel/standalone.html"
             else:
                 template = "caravel/explore.html"
-
-            resp = self.render_template(
-                template, viz=obj, slice=slc, datasources=datasources,
+            return self.render_template(
+                template, viz=viz_obj, slice=slc, datasources=datasources,
                 can_add=slice_add_perm, can_edit=slice_edit_perm,
                 can_download=slice_download_perm,
                 userid=g.user.get_id() if g.user else '')
 
-            try:
```

Comment:
oh wow, I'm not sure why we had that weird empty block here. Looks like a bad merge of some sort...